### PR TITLE
[Hub Generated] Review request for Microsoft.Advisor to add version stable/2020-01-01

### PR DESCRIPTION
**Pull request by yenchunchang**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/rebiex/azure-rest-api-specs/dev-advisor-Microsoft.Advisor-2020-01-01...master/).

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “**WaitForARMFeedback**” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [x] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Review by ChenTanyi — APPROVED** [review]

**Comment by ChenTanyi** [comment]
@rebiex please fix the `prettierCheck` and also add someone in your team for review.

**Comment by ChenTanyi** [comment]
@akning-ms @jianyexi Do we treat adding new response code as breaking change? In Java, it is not breaking.

**Comment by jianyexi** [comment]
> @akning-ms @jianyexi Do we treat adding new response code as breaking change? In Java, it is not breaking.

This causes API breaking change

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2020-01-01/advisor.json
Comment (rebiex) [inline specification/advisor/resource-manager/Microsoft.Advisor/stable/2020-01-01/advisor.json:549]: We are being flagged by S360 that our API would return 404 response code but we haven't define it in our Swagger. So I'm adding 404 response code here. https://portal.azure-devex-tools.com/amekpis/correctness/detail?errorId=46FCA6B0-7DA3-4527-8EE9-DADDF502FFAF

**Review by mudit794 — APPROVED** [review]

File: specification/advisor/resource-manager/Microsoft.Advisor/stable/2020-01-01/advisor.json
Comment (ChenTanyi) [reply specification/advisor/resource-manager/Microsoft.Advisor/stable/2020-01-01/advisor.json]: If so, I suggest you add `default` in every response rather than a single `404`. Otherwise, the other error response would also not appear in swagger, like `400`, `409`, `412`...

Comment (rebiex) [reply specification/advisor/resource-manager/Microsoft.Advisor/stable/2020-01-01/advisor.json]: You meant add default to our API controller or Swagger? If Swagger do you have an example for it?

Comment (ChenTanyi) [reply specification/advisor/resource-manager/Microsoft.Advisor/stable/2020-01-01/advisor.json]: You could see https://github.com/azure/azure-rest-api-specs/blob/a3733cc/specification/resources/resource-manager/Microsoft.Resources/stable/2020-06-01/resources.json#L49-L62

Comment (rebiex) [reply specification/advisor/resource-manager/Microsoft.Advisor/stable/2020-01-01/advisor.json]: Got it. For now we don't have a use case for other error code. Could we proceed with single 404 for now?

**Comment by rebiex** [comment]
Hi @ChenTanyi ,

If we want to procced with specific 404 error code is there anything that we are missing to get approval for this PR?

Thanks,
Yen-Chun

**Comment by ChenTanyi** [comment]
> Hi @rebiex, one or multiple breaking change(s) is detected in your PR. Please check out the breaking change(s), and provide business justification in the PR comment and @ PR assignee why you must have these change(s), and how external customer impact can be mitigated. Please ensure to follow [breaking change policy](https://aka.ms/AzBreakingChangesPolicy) to request breaking change review and approval before proceeding swagger PR review.
> If you think it is false positive breaking change, please provide the reasons in the PR comment, report to Swagger Tooling Team via https://aka.ma/swaggerfeedback.

@rebiex Please see this comment. You need to follow breaking change policy to get approval. So I suggest you to add all default response. Otherwise, you will need to go through it next time.

**Comment by rebiex** [comment]
> > Hi @rebiex, one or multiple breaking change(s) is detected in your PR. Please check out the breaking change(s), and provide business justification in the PR comment and @ PR assignee why you must have these change(s), and how external customer impact can be mitigated. Please ensure to follow [breaking change policy](https://aka.ms/AzBreakingChangesPolicy) to request breaking change review and approval before proceeding swagger PR review.
> > If you think it is false positive breaking change, please provide the reasons in the PR comment, report to Swagger Tooling Team via https://aka.ma/swaggerfeedback.
> 
> @rebiex Please see this comment. You need to follow breaking change policy to get approval. So I suggest you to add all default response. Otherwise, you will need to go through it next time.

Got it. Sorry I misunderstood at the first place. But I have question about why this would be a breaking change. We are not actually adding a new response code to our service we have 404 response code in our service already. The reason we are adding a new response code to Swagger is that we are being flagged for Swagger Correctness for having 404 response code in actual service call but not documented in Swagger. [Detail link](https://portal.azure-devex-tools.com/amekpis/correctness/detail?errorId=46FCA6B0-7DA3-4527-8EE9-DADDF502FFAF) Would this still be treated as breaking change?

**Comment by ChenTanyi** [comment]
Breaking change is considered by the change of swagger files instead of the actual behavior, since all SDK are generated by swagger files.

In this example, we don't care about whether service has returned 404 or not. Currently we treat 404 is an unexpected response, and after this change merged, it would become an expected response.

**Comment by rebiex** [comment]
Thanks for the clarification!

**Comment by rebiex** [comment]
Pending breaking change review https://msazure.visualstudio.com/One/_workitems/edit/8741257

**Review by JeffreyRichter — APPROVED** [review]

**Pull request merged**
